src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
```

Comment:
The variable name `currentState` is not fully indicative of its purpose. A name like `stateCopy` would better convey that this variable is a copy of the current state at each stage of transformation.